### PR TITLE
[Refactor] Remove alpha/beta rowset type

### LINKED ISSUE #7356: Remove alpha/beta rowset related concepts, just use rowset term directly.
## Refactoring request

alpha/beta rowset concept can be removed since there is no alpha rowset any longer.

- [x] Remove alpha/beta rowset type #7357
- [ ] Refactor beta rowset implementation into rowset, remove the inheritant hierarchy.
- [ ] Refactor beta rowset writer implementation into rowset writer, remove the inheritant hierarchy.

### PR DESCRIPTION
**Pull request by Linkerist**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7356

## Problem Summary(Required) ：

remove judgement about rowset type firstly.

### CODE REVIEW

**Comment by chaoyli** [comment]
Also the rowset type in the FE code

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by Linkerist** [comment]
> You can create a set of tasks in the #7356 issue.
> 
> 1. Remove the judgment in the BE code paths
> 2. Remove the judgment in the FE code paths.
> 3. Refactor the code to remove beta_rowset.cpp
> 4. Modify the meta when compaction and creating table.

done.

**Review by chaoyli — DISMISSED** [review]

File: gensrc/proto/olap_file.proto [hunk 1]
```diff
@@ -86,7 +86,7 @@ message RowsetMetaPB {
     optional int64 txn_id = 4;
     // tablet id and schema hash could find tablet
     optional int32 tablet_schema_hash = 5;
-    optional RowsetTypePB rowset_type = 6;
+    optional RowsetTypePB rowset_type = 6; // deleted
```
Comment (sduzh) [inline gensrc/proto/olap_file.proto:89]:
```suggestion
```

File: gensrc/proto/olap_file.proto [hunk 2]
```diff
@@ -244,7 +244,7 @@ message TabletMetaPB {
     // a uniqued id to identified tablet with same tablet_id and schema hash
     optional PUniqueId tablet_uid = 14;
     optional int64 end_rowset_id = 15;
-    optional RowsetTypePB preferred_rowset_type = 16;
+    optional RowsetTypePB preferred_rowset_type = 16; // deleted
```
Comment (sduzh) [inline gensrc/proto/olap_file.proto:247]:
```suggestion
    // optional RowsetTypePB preferred_rowset_type = 16;
```

**Comment by sduzh** [comment]
The `rowset_type` should be kept to support version rollback.

**Comment by Linkerist** [comment]
run starrocks_fe_unittest

**Comment by Linkerist** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by Linkerist** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by decster — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**